src/core/tests/type_prop/shape_of.cpp

```diff
@@ -137,3 +137,18 @@ TEST(type_prop, shape_of_3_dynamic_value_and_label_propagation) {
     const auto& output_shape = bc->get_output_partial_shape(0);
     ASSERT_EQ(ov::DimensionTracker::get_label(output_shape[0]), 10);
 }
+
+TEST(type_prop, shape_of_3_dynamic_value_propagation_out_i32) {
+    constexpr auto i32_max = static_cast<int64_t>(std::numeric_limits<int32_t>::max());
+    constexpr auto bound = static_cast<int64_t>(std::numeric_limits<int32_t>::max()) + 10;
+
+    auto param = std::make_shared<ov::op::v0::Parameter>(
+        element::f32,
+        PartialShape{{2, bound}, {3, -1}, {i32_max + 1, bound}, {i32_max, -1}, {1, 1021}});
+    auto op = std::make_shared<op::v3::ShapeOf>(param, element::i32);
+
+    auto bc_param = std::make_shared<ov::op::v0::Parameter>(element::f32, Shape{1});
+    auto bc = std::make_shared<op::v1::Broadcast>(bc_param, op);
+
+    EXPECT_EQ(bc->get_output_partial_shape(0), PartialShape({-1, -1, -1, -1, {1, 1021}}));
```

Comment:
why {3, -1} is a fully dynamic dimension? -1 is a special character always -- it means that upper bound is max_t

Reply:
Additionally, we may need to align the behavior of ShapeOf value propagation and PartialShape ctor.

Reply:
Ticket and test PR #21414 created to investigate issues when lower bound is not set to zero when upper is infinite.